### PR TITLE
Cd issue261 c

### PR DESCRIPTION
**Pull request by cdietrich**
@szarnekow here a variant with concurrent hashmap, null object and strings instead of fqn.
am not sure if i like it

### CODE REVIEW

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend [hunk 1]
```diff
@@ -12,30 +12,41 @@ import org.eclipse.xtext.common.types.descriptions.EObjectDescriptionBasedStubGe
...
+	Map<String, NameEnvironmentAnswer> cache = newHashMap()
     
 	override cleanup() {
 		cache.clear
 	}
 
 	override findType(char[][] compoundTypeName) {
 		val className = QualifiedName.create(compoundTypeName.map[String.valueOf(it)])
-		return findType(className)
+		return findType(className.toString)
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend]: No need to convert it to a QN in the first place.

Comment (cdietrich) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend]: how to easily do the join then?

Comment (cdietrich) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend]: list.join(".") ?

Comment (cdietrich) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend]: i will add the concurrent hashmap to the other one

Comment (szarnekow) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend]: org.eclipse.jdt.core.compiler.CharOperation.toString(char[][])

Comment (cdietrich) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/IndexAwareNameEnvironment.xtend]: we should decide to go for one of the two appoaches

**Comment by cdietrich** [comment]
@szarnekow is it ok if we close this one and follow the QN approach in the other pr?

**Pull request closed without merging**